tests/benchmarks/peripheral_stress_test/src/timer_temp_wdt_thread.c

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(timer, LOG_LEVEL_INF);
+
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/drivers/watchdog.h>
+#include "common.h"
+
+static struct k_timer my_timer;
+static int timer_expire_count;
+
+/* Temp sensor */
+static const struct device *temp_dev = DEVICE_DT_GET(DT_NODELABEL(temp_sensor));
+static enum sensor_channel chan_to_use = SENSOR_CHAN_DIE_TEMP;
+
+/* Watchdog */
+extern const struct device *const my_wdt_device;
+extern int my_wdt_channel;
+
+void my_work_handler(struct k_work *work)
+{
+	int rc;
+	struct sensor_value val;
+	int32_t temp_val;
+
+	/* Feed watchdog */
+	rc = wdt_feed(my_wdt_device, my_wdt_channel);
+	if (rc < 0) {
+		LOG_ERR("wdt_feed() returned: %d", rc);
```

Comment:
return after log? to detect failure?

Reply:
It's a work handler. I can return from it earlier but that will not work as You expect. I assume that WDT will reset cpu if feed() fails.